Public/Src/App/Bxl/Args.cs

```diff
@@ -1215,7 +1218,15 @@ public bool TryParse(string[] args, PathTable pathTable, out ICommandLineConfigu
                     // profile redirection only happens on Windows
                     layoutConfiguration.RedirectedUserProfileJunctionRoot = AbsolutePath.Invalid;
                 }
-                
+
+                if (OperatingSystemHelper.IsUnixOS)
+                {
+                    // TODO: Non Windows OS doesn't support admin-required process external execution mode.
+                    if (sandboxConfiguration.AdminRequiredProcessExecutionMode != AdminRequiredProcessExecutionMode.Internal)
+                    {
+                        throw CommandLineUtilities.Error(Strings.Args_AdminRequiredProcessExecutionMode_NotSupportedOnNonWindows, sandboxConfiguration.AdminRequiredProcessExecutionMode.ToString());
+                    }

```

Comment:
What's going to happen for our test pips that require admin on Windows but don't on macOS? For instance the integration test assembly runs incremental scheduling tests on Windows and needs admin but those test cases don't apply to macOS

Reply:
When running on macOS, currently one should not change the execution mode, and so pips will be run as they are today. Yes, some tests are only applied to windows. For inc. scheduling tests, all of them are marked `requiresJournalScan: true` which make them get filtered out when running on macOS. Later, if we need to address low-privilege on macOS, we can enrich the `privilegeLevel` enum in the pip specification. For example: `standard | adminWin | suUnix`.

---
In reply to: [277877974](https://github.com/Microsoft/BuildXL/pull/159#discussion_r277877974) [](ancestors = 277877974)